### PR TITLE
set CUDA target architecture(s) for GROMACS based on cuda_cc_semicolon_sep template value

### PR DESCRIPTION
**Pull request by jhmeinke**

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
Test report by @SebastianAchilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _GROMACS-2019.3-fosscuda-2019b.eb_ 
 * **SUCCESS** _GROMACS-2020.3-fosscuda-2019b.eb_ 
 * **SUCCESS** _GROMACS-2020.5-fosscuda-2020a-Python-3.8.2.eb_ 
 * **SUCCESS** _GROMACS-2021-fosscuda-2020b.eb_ 

Build succeeded for 4 out of 4 (4 easyconfigs in total)
cnx2 - Linux rocky linux 8.4, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8
See https://gist.github.com/9e041f92541c0ac44ad78db7b2448f4a for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @SebastianAchilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _GROMACS-2018.1-foss-2018b-PLUMED.eb_ 
 * **SUCCESS** _GROMACS-2018.2-foss-2018b.eb_ 
 * **SUCCESS** _GROMACS-2018.3-foss-2018b.eb_ 
 * **SUCCESS** _GROMACS-2018.4-foss-2018b-PLUMED-2.5.0.eb_ 
 * **SUCCESS** _GROMACS-2019.3-foss-2019a.eb_ 
 * **SUCCESS** _GROMACS-2019.3-foss-2019b.eb_ 
 * **SUCCESS** _GROMACS-2019.3-fosscuda-2019b.eb_ 
 * **SUCCESS** _GROMACS-2019.4-foss-2019b.eb_ 
 * **SUCCESS** _GROMACS-2019-foss-2018b.eb_ 
 * **FAIL (build issue)** _GROMACS-2020.1-foss-2020a-Python-3.8.2.eb_ (partial log available at https://gist.github.com/b7b9f00eaca7b634ebe07b9bf27f4e26)
 * **SUCCESS** _GROMACS-2020.3-fosscuda-2019b.eb_ 
 * **SUCCESS** _GROMACS-2020.4-foss-2020a-Python-3.8.2.eb_ 
 * **SUCCESS** _GROMACS-2020.5-fosscuda-2020a-Python-3.8.2.eb_ 
 * **SUCCESS** _GROMACS-2020-foss-2019b.eb_ 
 * **SUCCESS** _GROMACS-2021.3-foss-2021a-CUDA-11.3.1.eb_ 
 * **SUCCESS** _GROMACS-2021.3-foss-2021a-CUDA-11.3.1-PLUMED-2.7.2.eb_ 
 * **SUCCESS** _GROMACS-2021.3-foss-2021a.eb_ 
 * **SUCCESS** _GROMACS-2021-foss-2020b.eb_ 
 * **SUCCESS** _GROMACS-2021-fosscuda-2020b.eb_ 

Build succeeded for 18 out of 19 (19 easyconfigs in total)
cnx1 - Linux rocky linux 8.4, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/19b915b93e4597043bb78702e42ce7ba for a full test report.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/g/gromacs.py [hunk 1]
```diff
@@ -190,9 +190,13 @@ def configure_step(self):
                     self.cfg.update('configopts', "-DGMX_GPU=CUDA -DCUDA_TOOLKIT_ROOT_DIR=%s" % cuda)
                 else:
                     self.cfg.update('configopts', "-DGMX_GPU=ON -DCUDA_TOOLKIT_ROOT_DIR=%s" % cuda)
+
+                # Set CUDA capabilities based on template value.
+                cuda_cc_semicolon_sep = self.cfg.get_cuda_cc_template_value("cuda_cc_semicolon_sep").replace('.', '')
+                self.cfg.update('configopts', '-DGMX_CUDA_TARGET_SM="%s"' % cuda_cc_semicolon_sep)
```
[the remark this answers is not in the document]
Comment (jhmeinke) [reply easybuild/easyblocks/g/gromacs.py]: Good point. Implemented in be6c3e8.

File: easybuild/easyblocks/g/gromacs.py [hunk 2]
```diff
@@ -190,9 +190,14 @@ def configure_step(self):
                     self.cfg.update('configopts', "-DGMX_GPU=CUDA -DCUDA_TOOLKIT_ROOT_DIR=%s" % cuda)
                 else:
                     self.cfg.update('configopts', "-DGMX_GPU=ON -DCUDA_TOOLKIT_ROOT_DIR=%s" % cuda)
+
+                # Set CUDA capabilities based on template value.
+                if '-DGMX_CUDA_TARGET_SM' not in self.cfg['configopts']:
+                    cuda_cc_semicolon_sep = self.cfg.get_cuda_cc_template_value("cuda_cc_semicolon_sep").replace('.', '')
```
[the remark this answers is not in the document]
Comment (jhmeinke) [reply easybuild/easyblocks/g/gromacs.py:196]: Fixed in 5e4c9ef.

Comment (jhmeinke) [reply easybuild/easyblocks/g/gromacs.py]: I hope I got the correct line this time (see 930c24a).

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _GROMACS-2021.2-fosscuda-2020b.eb_ 
 * **SUCCESS** _GROMACS-2020.3-fosscuda-2019b.eb_ 
 * **SUCCESS** _GROMACS-2021.3-foss-2021a-CUDA-11.3.1-PLUMED-2.7.2.eb_ 
 * **SUCCESS** _GROMACS-2021-foss-2020b.eb_ 

Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3306.joltik.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), 1 x NVIDIA Tesla V100-SXM2-32GB, 510.47.03, Python 3.6.8
See https://gist.github.com/d4cb6b9ccc97a7f3d09198c2f27c7764 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _GROMACS-2021.2-fosscuda-2020b.eb_ 
 * **SUCCESS** _GROMACS-2021.3-foss-2021a-CUDA-11.3.1-PLUMED-2.7.2.eb_ 
 * **SUCCESS** _GROMACS-2021.5-foss-2021b.eb_ 
 * **SUCCESS** _GROMACS-2021.5-foss-2021b-CUDA-11.4.1.eb_ 

Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3901.accelgor.os - Linux RHEL 8.4, x86_64, AMD EPYC 7413 24-Core Processor (zen3), 1 x NVIDIA NVIDIA A100-SXM4-80GB, 510.39.01, Python 3.6.8
See https://gist.github.com/ced9639ed8bc9e10ae3fc59deb001bde for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Looks good, tested extensively, so going in, thanks a lot for the contribution @jhmeinke!

**Pull request merged**